### DISCUSSION TITLE
[Bug] The traceId get from logstash  are all mixed up accross multiple tomcat threads

### DISCUSSION THREAD

**Discussion by whohasthis** [apache/skywalking#7809] [Q&A]
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/skywalking/issues?q=is%3Aissue) and found no similar issues.

### Apache SkyWalking Component

Java Agent (apache/skywalking-java)

### What happened

We use TraceIdJsonProvider of apm-toolkit-logback-1.x to add the trace id into the logstash fields sent to ELK,  we are expecting to use the trace id to get all related log message throughout all services that the trace id involves. 

Below is the logback config part related to the trace id logstash appender

<appender name="logstash"
        class="net.logstash.logback.appender.LogstashTcpSocketAppender">
        <destination>logstash.:5045</destination>
        <encoder class="net.logstash.logback.encoder.LogstashEncoder">
            <customFields>
                {
                "service" : {
                "category" : "xxx",
                "type" : "xxxxx",
                "name" : "xxx-xxx-xx"
                }
                }
            </customFields>
            <provider class="org.apache.skywalking.apm.toolkit.log.logback.v1.x.logstash.TraceIdJsonProvider" />
        </encoder>
    </appender>

But somehow the trace ids are all mixed up, we are seeing that one trace id associated to many different tomcat thread name  in the same short period of time, below are part of the result we found on ELK

![elk-traceid](https://user-images.githubusercontent.com/61497460/134808657-54b1a706-d8f1-42a7-859e-1d1e3889e624.png)

### What you expected to happen

We expect to get one trace id per one thread name during the same period of time, not all mixed up accoss many different thread names.

### How to reproduce

We use skywalking 8.4.0, OS centos 6.7, 
Skywalking OAP server is running on K8S env, using ES 6.7 as the storage.
The service is running on tomcat 8.5.65 with spring MVC framework.

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

**Comment by wu-sheng** [comment]
If trace id gets mixed, it means you are facing some plugin side issue. You need to locate what it is, and why(not supported version, not supported cast).

**Reply by wu-sheng** [reply]
Tomcat plugin has little chance to have this issue. Its multiple threads never reused before the request finished.
